pricing.js

```diff
@@ -1,37 +1,45 @@
+function formatPrice(price) {
+  return parseInt(price * 100) / 100
+}
+
+function calculateEmployerContribution(product, price, dollarsOff) {
+  if (product.employerContribution.mode === 'dollar') {
+    price = price - product.employerContribution.contribution
+  } else {
+    dollarsOff = price * (product.employerContribution.contribution / 100)
+    price = price - dollarsOff
+  }
+}
+
+function CalculateVolLifePrice(price, product, coverageLevels) {
+  for (var i = 0; i < coverageLevels.length; i++) {
+    var coverageAmount = coverageLevels[i].coverage
+
+    price += (coverageAmount / product.cost.costDivisor) * product.cost.price
+  }
+  return price
+}
+
+function calculateLTDPrice(product, employee) {
+  var salaryPercentage = product.coveragePercentage / 100
+
+  return ((employee.salary * salaryPercentage) / product.cost.costDivisor) * product.cost.price
+}
 module.exports.calculateProductPrice = function (product, employee, coverageLevels) {
   var price = 0
   var dollarsOff = 0
-
+  var DiscountedPrice
   switch (product.type) {
     case 'volLife':
-      for (var i = 0; i < coverageLevels.length; i++) {
-        var coverageAmount = coverageLevels[i].coverage
-
-        price += (coverageAmount / product.cost.costDivisor) * product.cost.price
-      }
-
-      if (product.employerContribution.mode === 'dollar') {
-        price = price - product.employerContribution.contribution
-      } else {
-        dollarsOff = price * (product.employerContribution.contribution / 100)
-        price = price - dollarsOff
-      }
-
-      return parseInt(price * 100) / 100
+      price = CalculateVolLifePrice(price, dollarsOff, product, coverageLevels)
+      DiscountedPrice = calculateEmployerContribution(product, price, dollarsOff)
+      return formatPrice(DiscountedPrice)
     case 'ltd':
-      var salaryPercentage = product.coveragePercentage / 100
-
-      price += ((employee.salary * salaryPercentage) / product.cost.costDivisor) * product.cost.price
-
-      if (product.employerContribution.mode === 'dollar') {
-        price = price - product.employerContribution.contribution
-      } else {
-        dollarsOff = price * product.employerContribution.contribution
-        price = price - dollarsOff
-      }
-
-      return parseInt(price * 100) / 100
+      price = calculateLTDPrice(product, employee)
+      DiscountedPrice = calculateEmployerContribution(product, price, dollarsOff)
+      return formatPrice(DiscountedPrice)
+      return calculateLTDPrice(product, employee)
```

Comment:
There are two issues with line 41.

The first is that we are calculating the LTD price twice. Line 38 calculates the price, line 39 figures out the discounted price and line 40 formats the final price and sends it back. We are then saying on line 41 that we want to calculate the price a second time.

The second is that it is technically unreachable (VS Code grays it out slightly to show this) because on line 40 we say `return formatPrice(DiscountedPrice)`. That means when we get to line 40 we leave this function as we `return` the value back to the caller. Since we will always leave the function on line 40, we could never run line 41.